cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -197,7 +197,11 @@ class device_row_comparator {
           return cuda::std::make_pair(state, depth);
         }
 
-        // Structs have been modified to only have 1 child when using this.
+        if (lcol.num_child_columns() == 0) {
+          return cuda::std::make_pair(weak_ordering::EQUIVALENT, depth);
+        }
```

Comment:
@devavret I assume there is some check earlier on in hose code that verifies corresponding struct columns have the same number of children?

Reply:
Yes. If it reaches here, they're the same structure.

Reply:
> I didn't know there could be empty structs. Does this work if a struct contains 2 struct children, the first of which is empty?

It does work. I added a test case to cover this scenario. IIUC, following the rule of `decompose_structs`, the second child will be detached from the root struct.

```
     S0
     / \
  S1  S2
```
=>
```
S0
 |
S1   S2
```